lib/tasks/multitenant/tenants.rake

```diff
@@ -77,6 +77,49 @@ namespace :multitenant do
       Rails.logger.error "Inconsistent tenant_ids for:\n#{inconsistent.map {_1.join(" ")}.join("\n")}"
     end
 
+    desc 'Check and remove orphaned objects (whose tenant is missing), pass "destroy" argument to delete'
+    task :cleanup_orphans, [:mode] => :environment do |_task, args|
+      destroy = args[:mode] == "destroy"
+
+      puts "Checking orphaned objects..."
+      puts "WARNING: the found orphan objects will be destroyed" if destroy
+
+      provider_account_ids = Account.where(provider: true).pluck(:id) + [Account.master.id]
+
+      base_models.each do |model|
+        orphaned_objects = model.where.not(tenant_id: provider_account_ids)
+
+        if orphaned_objects.exists?
+          puts "Found #{orphaned_objects.size} orphaned objects for model #{model.name}:"
+          orphaned_objects.find_in_batches(batch_size: 100).with_index do |batch, index|
```

Comment:
seems like [find_each](https://apidock.com/rails/ActiveRecord/Batches/find_each) will simplify this

Reply:
The idea (that I haven't finished pushing, it seems), is to add some "sleep" time between each batch, to space the deletion jobs.

Reply:
The whole thing should happen in a different way. Most likely background deletion. Optimize this within a single rake task will probably be fragile in any case. If you want leave this alone until we figure out background deletion and then we may have a better method to handle this.